### PR TITLE
fix: check if registrationEmailAsUsername is explicitly set to false

### LINKED ISSUE #27137: Users with fine-grained permissions can not create a user
### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

admin/ui

### Describe the bug

Users with fine-grained permissions that do have the manage-users permissions for a group, but not the manage-users role, can not create users.

The problem is with
```
isDisabled={
            !user?.id &&
            !watchUsernameInput &&
           !realm.registrationEmailAsUsername
          }
```
The user does not have rights to view `realm.registrationEmailAsUsername` and since the other  2 fields are also `undefined` (username field is not visible since email as username is enabled and since we're creating a user `user.id` is also `undefined`) 

### Version

nightly

### Regression

- [ ] The issue is a regressions

### Expected behavior

The user should be able to create a user

### Actual behavior

The create button is disabled

### How to Reproduce?

1. Create a user that has the `query-users` and `query-groups` role
2. Create a group 
3. Enable fine grained permissions on that group
4. Enable the fine grained `manage-members` and `manage-membership` on the group for the created user
5. Impersonate the user
6. Try to create a user

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by pkeuter**
…instead of falsey

Closes #27137

### CODE REVIEW

File: js/.husky/pre-commit [hunk 1]
```diff
@@ -1,2 +1,3 @@
+#!/bin/sh
```
Comment (jonkoops) [inline js/.husky/pre-commit:1]: This change seems unrelated, and the docs for husky state that this will always execute with `sh`, so there should be no need to restore this. Let's remove it here.

**Review by jonkoops — CHANGES_REQUESTED** [review]

File: js/.husky/pre-commit (hunk 1 above)
Comment (pkeuter) [reply js/.husky/pre-commit:1]: Husky throws an error when this is not here. 

See here: https://github.com/conventional-changelog/standard-version/issues/254#issuecomment-424669323

This is conventional-changelog, but the error is about husky. Without this line, I am unable to commit.

Comment (pkeuter) [reply js/.husky/pre-commit:1]: Might be because I am on Windows?

**Review by ssilvert — COMMENTED** [review]
I approve this as long as the concern from @jonkoops is addressed.  Assigning this PR to him.

File: js/.husky/pre-commit (hunk 1 above)
Comment (pkeuter) [reply js/.husky/pre-commit]: But you are right. This change does not belong in this PR so I've removed it.

**Comment by pkeuter** [comment]
I have removed the pre-commit fix so this should be ready to go now

**Review by jonkoops — APPROVED** [review]
LGTM, let's get this merged.

**Comment by jonkoops** [comment]
Thanks for the contribution @pkeuter!
